core/app/models/spree/line_item.rb

```diff
@@ -118,7 +124,10 @@ def options=(options = {})
 
       assign_attributes options
 
-      # There's no need to call a pricer if we'll set the price directly.
+      # When price is part of the options we are not going to fetch
+      # it from the variant. Please note that this always allows to set
+      # a price for this line item, even if there is no existing price
+      # for the associated line item in the order currency.
```

Comment:
Really good documentation. I wish I'd had that when first reading this method :)